File: 2308-divide-array-into-equal-pairs/2308-divide-array-into-equal-pairs.cpp
class Solution {
public:
    bool divideArray(vector<int>& nums) {
        map<int, int> res;

        for (int i = 0; i < nums.size(); i++) {

            res[nums[i]]++;
        }

        for (auto i : res) {
            if (i.second % 2 != 0)
                return false;
        }
        return true;
    }
};

File: 2308-divide-array-into-equal-pairs/2308-divide-array-into-equal-pairs.java
class Solution {
    public boolean divideArray(int[] nums) {//using freq array withing range given 1<= n <=500
        int freq[] = new int[501];
        for (int i = 0; i < nums.length; i++) {
            freq[nums[i]]++;

        }

        for (int i = 1; i < freq.length; i++) {
            if (freq[i] % 2 != 0)
                return false;

        }

        return true;
    }
}